redfish/provider/resource_redfish_storage_controller.go

```diff
@@ -0,0 +1,763 @@
+/*
+Copyright (c) 2024 Dell Inc., or its subsidiaries. All Rights Reserved.
+
+Licensed under the Mozilla Public License Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://mozilla.org/MPL/2.0/
+
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"terraform-provider-redfish/common"
+	"terraform-provider-redfish/gofish/dell"
+	"terraform-provider-redfish/redfish/models"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/stmcginnis/gofish"
+	redfishcommon "github.com/stmcginnis/gofish/common"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource = &RedfishStorageControllerResource{}
+)
+
+// NewRedfishStorageControllerResource is a helper function to simplify the provider implementation.
+func NewRedfishStorageControllerResource() resource.Resource {
+	return &RedfishStorageControllerResource{}
+}
+
+// RedfishStorageControllerResource is the resource implementation.
+type RedfishStorageControllerResource struct {
+	p *redfishProvider
+}
+
+// Configure implements resource.ResourceWithConfigure
+func (r *RedfishStorageControllerResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	r.p = req.ProviderData.(*redfishProvider)
+}
+
+// Metadata returns the resource type name.
+func (*RedfishStorageControllerResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "storage_controller"
+}
+
+// Schema defines the schema for the resource.
+func (*RedfishStorageControllerResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "This Terraform resource is used to configure the storage controller. " +
```

Comment:
please add this resource supported terraform cmd, such as import, update and others